core/src/main/java/org/testcontainers/containers/wait/strategy/AbstractWaitStrategy.java

```diff
@@ -6,10 +6,26 @@
 
 import java.time.Duration;
 import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
 
 public abstract class AbstractWaitStrategy implements WaitStrategy {
 
+    static final ExecutorService EXECUTOR = Executors.newCachedThreadPool(new ThreadFactory() {
```

Comment:
Could we use Guava's `ThreadFactoryBuilder` here?

Reply:
I would rather reduce Guava usage, not increase it 😅 I _could_ change it, but this is some "static" code, and it is not huge, so... WDYT? :)

Reply:
I'd say not use Guava and use a real class `WaitStrategyDaemonThreadFactory` or something.

Reply:
what's wrong with a non-reusable 10 lines anonymous class close to the usage? 😅

Reply:
That I am confronted with lower-level implementation detail at the head of the file instead of a class name acting as an abstraction. But these are differences in our style, which we might never overcome 😅

Reply:
I don't mind; I'd rather see the removal of Guava as a dependency too, but it's there.

Perhaps we could extract this to our own class, a bit like @kiview suggests? We have the same code over in `Startables`.